### PR TITLE
Add k8s 1.25.2

### PR DESCRIPTION
**Pull request by jiaqiluo**
Issue: https://github.com/rancher/rancher/issues/38701

This PR does the following things:
- add k8s v1.25.2-rancher1-1
- add the template for Canal 3.24.1
- add the template for Calico 3.24.1
- add the template for Flannel 0.19.2

Tests:
The PR is tested using a customized RKE in which the KDM default URL is set to this branch to provision single node clusters with k8s `v1.25.2-rancher1-1` and each of those three new networks plugin templates. 
The provision of all those clusters succeeds.

### CODE REVIEW

File: rke/k8s_version_info.go
Comment (a-blender) [inline rke/k8s_version_info.go:647]: Why is the `MinRancherVersion` a patch0?

File: rke/k8s_version_info.go
Comment (a-blender) [inline rke/k8s_version_info.go:648]: There is no rke 1.4.0-rc0. Why are we using that? The latest version of rke is 1.4.0-rc4 https://github.com/rancher/rke/tags. Is the general policy to use an rc here for the min version or what docs say what to use for 1.25?

**Review by a-blender — DISMISSED** [review]
Had a few questions @jiaqiluo, otherwise looks great

**Comment by a-blender** [comment]
@jiaqiluo Is this DNM for 2.7.0?

File: rke/templates/flannel_v0.19.2.go
Comment (kinarashah) [inline rke/templates/flannel_v0.19.2.go]: We should carry forward {{.NodeSelector}} changes, it's a rancher specific change and comes from rke.

File: rke/k8s_version_info.go
Comment (kinarashah) [reply rke/k8s_version_info.go:648]: It's a general pattern we follow, we need to use RCs so the version loads for testing. 
@jiaqiluo let's make sure this doesn't land into the RKE version which will get tagged with out of band KDM release for November.

File: rke/k8s_version_info.go
Comment (kinarashah) [reply rke/k8s_version_info.go:647]: We've occasionally tagged `-patch` versions after Rancher releases for specific bug fixes. Kubernetes versions should still load for them same as `v2.7.1` so we set it as min version beforehand than having to make a change later on.

**Review by kinarashah — COMMENTED** [review]
Small changes. @jiaqiluo Canal and calico versions are v3.24.1 but commits say v2.34.1, could we fix the typo? 
Also thanks for splitting the commits neatly, was easy to review! Let's squash the later commits for template and image changes before merging.

File: rke/k8s_version_info.go
Comment (jiaqiluo) [reply rke/k8s_version_info.go:648]: > @jiaqiluo let's make sure this doesn't land into the RKE version which will get tagged with out of band KDM release for November.

No problem, I will hold this PR.

File: rke/templates/flannel_v0.19.2.go
Comment (jiaqiluo) [reply rke/templates/flannel_v0.19.2.go]: The upstream chart adopts the `{{.NodeSelector}` section, so I removed this section to reduce duplicates.  

see line 169-174

**Comment by jiaqiluo** [comment]
hi @annablender @kinarashah  
The PR is ready for your final review. All your comments are addressed/answered.

**Review by a-blender — DISMISSED** [review]
@kinarashah Thanks for clarifying on the minVersion and patch release, that was helpful. @jiaqiluo are you sure you mean DNM Rancher 2.6.1 in the title? And could you also update the description to 3.24.1 canal and calico versions? Otherwise this LGTM

File: rke/k8s_rke_system_images.go
Comment (kinarashah) [inline rke/k8s_rke_system_images.go:8378]: need comment on top of this, `Enabled in Rancher vx.x.x`

File: rke/k8s_rke_system_images.go
Comment (kinarashah) [inline rke/k8s_rke_system_images.go]: a new mirrored metrics-server is available, v0.6.2...could you update the image?

File: rke/k8s_rke_system_images.go
Comment (jiaqiluo) [reply rke/k8s_rke_system_images.go:8378]: comment is added

File: rke/k8s_rke_system_images.go
Comment (jiaqiluo) [reply rke/k8s_rke_system_images.go]: updated.

**Review by Oats87 — DISMISSED** [review]
LGTM; @kinarashah needs to review before we can merge though.

File: rke/k8s_version_info.go
Comment (kinarashah) [inline rke/k8s_version_info.go:647]: @jiaqiluo would this be `2.7.2-patch0` if we're enabling with v2.7.2?

Comment (kinarashah) [reply rke/k8s_version_info.go:647]: discussed offline, the comment `// Enabled in Rancher v2.7.2` needs to be v2.7.1.

**Review by a-blender — DISMISSED** [review]
Make sure to address this comment https://github.com/rancher/kontainer-driver-metadata/pull/991/files#r1040185335, otherwise LGTM

File: rke/k8s_version_info.go
Comment (jiaqiluo) [reply rke/k8s_version_info.go:647]: fixed

**Review by kinarashah — APPROVED** [review]

**Pull request merged**
